tests/phpunit/tests/blocks/insertHookedBlocks.php

```diff
@@ -0,0 +1,203 @@
+			array( self::HOOKED_BLOCK_TYPE ),
+			$anchor_block['attrs']['metadata']['ignoredHookedBlocks'],
+			"ignoredHookedBlocks metadata shouldn't have been modified."
+		);
+		$this->assertSame(
+			'',
+			$actual,
+			"No markup should've been generated for ignored hooked block."
+		);
+	}
+
+	/**
+	 * @ticket 59572
+	 * @ticket 60126
+	 *
+	 * @covers ::insert_hooked_blocks
+	 */
+	public function test_insert_hooked_blocks_adds_to_ignored_hooked_blocks() {
+		$anchor_block = array(
+			'blockName' => 'tests/anchor-block',
+			'attrs'     => array(
+				'metadata' => array(
+					'ignoredHookedBlocks' => array( self::HOOKED_BLOCK_TYPE ),
+				),
+			),
+		);
+
+		$actual = insert_hooked_blocks( $anchor_block, 'before', self::HOOKED_BLOCKS, array() );
+		$this->assertSame(
+			array( self::HOOKED_BLOCK_TYPE, self::OTHER_HOOKED_BLOCK_TYPE ),
+			$anchor_block['attrs']['metadata']['ignoredHookedBlocks'],
+			"Newly hooked block should've been added to ignoredHookedBlocks metadata while retaining previously ignored one."
+		);
+		$this->assertSame(
+			'<!-- wp:' . self::OTHER_HOOKED_BLOCK_TYPE . ' /-->',
+			$actual,
+			"Markup for newly hooked block should've been generated."
+		);
+	}
+
+	/**
+	 * @ticket 59572
+	 * @ticket 60126
+	 *
+	 * @covers ::insert_hooked_blocks
+	 */
+	public function test_insert_hooked_blocks_filter_can_set_attributes() {
+		$anchor_block = array(
+			'blockName'    => self::ANCHOR_BLOCK_TYPE,
+			'attrs'        => array(
+				'layout' => array(
+					'type' => 'constrained',
+				),
+			),
+			'innerContent' => array(),
+		);
+
+		$filter = function ( $parsed_hooked_block, $relative_position, $parsed_anchor_block ) {
+			// Is the hooked block adjacent to the anchor block?
+			if ( 'before' !== $relative_position && 'after' !== $relative_position ) {
+				return $parsed_hooked_block;
+			}
+
+			// Does the anchor block have a layout attribute?
+			if ( isset( $parsed_anchor_block['attrs']['layout'] ) ) {
+				// Copy the anchor block's layout attribute to the hooked block.
+				$parsed_hooked_block['attrs']['layout'] = $parsed_anchor_block['attrs']['layout'];
+			}
+
+			return $parsed_hooked_block;
+		};
+		add_filter( 'hooked_block_' . self::HOOKED_BLOCK_TYPE, $filter, 10, 3 );
+		$actual = insert_hooked_blocks( $anchor_block, 'after', self::HOOKED_BLOCKS, array() );
+		remove_filter( 'hooked_block_' . self::HOOKED_BLOCK_TYPE, $filter, 10, 3 );
+
+		$this->assertSame(
+			array( self::HOOKED_BLOCK_TYPE ),
+			$anchor_block['attrs']['metadata']['ignoredHookedBlocks'],
+			"Hooked block type wasn't added to ignoredHookedBlocks metadata."
+		);
+		$this->assertSame(
+			'<!-- wp:' . self::HOOKED_BLOCK_TYPE . ' {"layout":{"type":"constrained"}} /-->',
+			$actual,
+			"Markup wasn't generated correctly for hooked block with attribute set by filter."
+		);
+	}
+
+	/**
+	 * @ticket 59572
+	 * @ticket 60126
+	 *
+	 * @covers ::insert_hooked_blocks
+	 */
+	public function test_insert_hooked_blocks_filter_can_wrap_block() {
```

Comment:
Excellent addition, these two tests cover new filters. It's exactly what I expected.